File: CSharpWebServer/CSharpWebServer/DOM/ComplianceTest.cs
using CSharpWebServer.ist.enesuysal.thesis;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharpWebServer.DOM
{
    public class ComplianceTest
    {
        public string GetMessage(string BASE64String)
        {
            //try
            //{
            byte[] decodedBytes = Convert.FromBase64String(BASE64String);
            StringBuilder sb = new StringBuilder();
            Receiver r = new Receiver();
            for (int i = 0; i < r.knownMethods.Count; i++)
            {

                if (r.knownMethods[i].myfields.Length == 1 && Helper.CheckPrimitive(decodedBytes))
                {
                    //Check Type
                    if (Helper.GetFieldType(decodedBytes[0]) == Helper.GetFieldType(r.knownMethods[i].myfields[0])) //Create New bytearray
                    {

                        sb.AppendLine(r.createPrimitive(decodedBytes));
                    }
                }
                else if (!Helper.CheckPrimitive(decodedBytes))
                {

                    bool found = false;
                    byte[] finalArray = new byte[0];
                    String className = "";
                    bool flag = false;
                    while (Helper.GetFieldBytes(r.knownMethods[i].myfields).Length != 0) {
                       // if (!flag)
                        found = false;
                        byte[] fieldByte = Helper.GetFieldBytes(r.knownMethods[i].myfields);
                       
                        if (fieldByte[0] == 1) {

                            byte[] newFieldByte = Helper.pop(fieldByte);
                            if (Helper.indexOf(decodedBytes, newFieldByte) != -1) {
                                found = true;
                                className = r.knownMethods[i].methodName;
                               // sb.AppendLine("Mandatotory field found @" + r.knownMethods[i].methodName);
                                finalArray = Helper.push(finalArray, newFieldByte);
                            } else {
                                //sb.AppendLine("Mandatotory field is not found");
                                break;
                            }
                        }else {
                            //fieldByte= Helper.pop(fieldByte);
                            byte[] newFieldByte = Helper.pop(fieldByte);
                            //System.out.println(Arrays.toString(decodedBytes));
                            //System.out.println(Arrays.toString(fieldByte));
                            if (Helper.indexOf(decodedBytes, newFieldByte) != -1) {
                                found = true;
                                className = r.knownMethods[i].methodName;
                                //sb.AppendLine("Optional field found with field name and value equlity @"+ r.knownMethods[i].methodName);
                                finalArray = Helper.push(finalArray, newFieldByte);
                            } else {

                                if (Helper.CheckOptional(decodedBytes, newFieldByte)) {
                                   // sb.AppendLine("Optional field found with field name equlity @" + r.knownMethods[i].methodName);
                                   // Console.WriteLine(Array.T.toString(newFieldByte));
                                    className = r.knownMethods[i].methodName;
                                    finalArray = Helper.push(finalArray, newFieldByte);
                                    
                                    found = true;
                                } else {
                                    //sb.AppendLine("Optional field is not found");
                                    if (r.createPrimitive(newFieldByte) != "")
                                    {
                                        found = true;
                                        finalArray = Helper.push(finalArray, newFieldByte);
                                        //break;
                                    }
                                  
                                }
                            }
                        }
                        r.knownMethods[i].myfields = Helper.pop(r.knownMethods[i].myfields, fieldByte);
                    }
                    if (found) {
                        sb.AppendLine("" + r.knownMethods[i].methodName);
                        sb.AppendLine("Class Values{");
                       //Console.WriteLine(Arrays.toString(finalArray));
                        sb.AppendLine(r.createObject(className, finalArray));
                        sb.AppendLine("}");
                         break;
                    }
                }

            }
            return sb.ToString();

            //}
            //catch (Exception ex)
            //{
            //    throw ex;
            //}
        }
    }
}

File: JavaWebServer/src/java/DOM/ComplianceTest.java
/*
 *  Information Systems and Computer Engineering
 *  Distributed programming in cloud computing
 *  Enes UYSAL
 *  Enes UYSAL
 *   
 *   Supervisor: José Carlos Martins Delgado
 */
package DOM;

import ist.enesuysal.thesis.Helper.Helper;
import ist.enesuysal.thesis.Receiver;
import java.io.IOException;
import java.util.Arrays;
import sun.misc.BASE64Decoder;

/**
 *
 * @author enes
 */
public class ComplianceTest {

    public String GetMessage(String BASE64String) throws ClassNotFoundException, Exception {
        BASE64Decoder decoder = new BASE64Decoder();
        StringBuilder sb = new StringBuilder();
        try {
            byte[] decodedBytes = decoder.decodeBuffer(BASE64String);
            
            
            Receiver r = new Receiver();
            outerloop:
            for (int i = 0; i < r.knownMethods.length; i++) {

                if (r.knownMethods[i].myfields.length == 1 && Helper.CheckPrimitive(decodedBytes)) {
                    //Check Type
                    if (decodedBytes[0] == r.knownMethods[i].myfields[0]) //Create New bytearray
                    {
                       sb.append(r.createPrimitive(decodedBytes));
                       sb.append(System.getProperty("line.separator"));
                    }
                } else if (!Helper.CheckPrimitive(decodedBytes)) {
                    //Object 
                    //System.out.println(Arrays.toString(r.knownMethods[i].myfields));
                    boolean found = false;
                    byte[] finalArray = new byte[0];
                    String className = "";
                    
                    while (Helper.GetFieldBytes(r.knownMethods[i].myfields).length != 0) {
                        found = false;
                        byte[] fieldByte = Helper.GetFieldBytes(r.knownMethods[i].myfields);
                       
                        if (fieldByte[0] == 1) {

                            byte[] newFieldByte = Helper.pop(fieldByte);
                            if (Helper.indexOf(decodedBytes, newFieldByte) != -1) {
                                found = true;
                                className = r.knownMethods[i].methodName;
                               // sb.append("Mandatotory field found @"+ r.knownMethods[i].methodName);
                                //sb.append(System.getProperty("line.separator"));
                                finalArray = Helper.push(finalArray, newFieldByte);
                            } else {
                               // sb.append("Mandatotory field is not found");
                                //sb.append(System.getProperty("line.separator"));
                                break;
                            }
                        } else {
                            //fieldByte= Helper.pop(fieldByte);
                            byte[] newFieldByte = Helper.pop(fieldByte);
                            //System.out.println(Arrays.toString(decodedBytes));
                            //System.out.println(Arrays.toString(fieldByte));
                            if (Helper.indexOf(decodedBytes, newFieldByte) != -1) {
                                found = true;
                                className = r.knownMethods[i].methodName;
                              //  sb.append("Optional field found with field name and value equlity @"+ r.knownMethods[i].methodName);
                                //sb.append(System.getProperty("line.separator"));
                                finalArray = Helper.push(finalArray, newFieldByte);
                            } else {

                                if (Helper.CheckOptional(decodedBytes, newFieldByte)) {
                                  // sb.append("Optional field found with field name equlity @"+ r.knownMethods[i].methodName);
                                    //sb.append(System.getProperty("line.separator")); 
                                    className = r.knownMethods[i].methodName;
                                    finalArray = Helper.push(finalArray, newFieldByte);
                                    
                                    found = true;
                                } else {
                                   // sb.append("Optional field is not found");
                                   // sb.append(System.getProperty("line.separator"));
                                   break;
                                }
                            }
                        }
                        r.knownMethods[i].myfields = Helper.pop(r.knownMethods[i].myfields, fieldByte);
                    }
                    if (found) {
                        sb.append("Found Object");
                        sb.append(System.getProperty("line.separator"));
                        sb.append(r.createObject(className, finalArray));
                        sb.append(System.getProperty("line.separator"));
                         break outerloop;
                    }
                }
            }

        } catch (IOException ex) {
            sb.append("The Error Occured " + ex.getMessage());
        }
        return sb.toString();
    }

    public boolean IsPrimitiveArraySame(byte[] array1, byte[] array2) {
        return Arrays.equals(array1, array2);
    }
}
